File: app/src/main/java/com/zerostress/manager/SettingsActivity.java
package com.zerostress.manager;

import android.content.Intent;
import android.os.Bundle;
import android.widget.Toast;

import androidx.appcompat.app.AlertDialog;
import androidx.appcompat.app.AppCompatActivity;

import com.google.android.material.button.MaterialButton;
import com.google.android.material.switchmaterial.SwitchMaterial;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.firestore.FirebaseFirestore;

public class SettingsActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_settings);

        MaterialButton btnClearCache = findViewById(R.id.btnClearCache);
        MaterialButton btnAbout = findViewById(R.id.btnAbout);
        MaterialButton btnLogout = findViewById(R.id.btnSettingsLogout);
        MaterialButton btnDeleteAccount = findViewById(R.id.btnDeleteAccount);
        SwitchMaterial switchNotifications = findViewById(R.id.switchNotifications);
        SwitchMaterial switchChatNotifs = findViewById(R.id.switchChatNotifications);
        SwitchMaterial switchScheduleNotifs = findViewById(R.id.switchScheduleNotifications);

        btnClearCache.setOnClickListener(v -> {
            Toast.makeText(this, "Cache cleared!", Toast.LENGTH_SHORT).show();
        });

        btnAbout.setOnClickListener(v -> {
            new AlertDialog.Builder(this)
                .setTitle("ZERO STRESS")
                .setMessage("Version 3.0\n\nPerformance & Leaderboard Manager\n\nBuilt with Firebase + Java")
                .setPositiveButton("OK", null)
                .show();
        });

        btnLogout.setOnClickListener(v -> {
            FirebaseAuth.getInstance().signOut();
            startActivity(new Intent(this, LoginActivity.class));
            finish();
        });

        // Delete account (2-step confirmation)
        btnDeleteAccount.setOnClickListener(v -> {
            new AlertDialog.Builder(this)
                .setTitle("⚠️ Delete Account")
                .setMessage("This will permanently delete your account and all your data.\n\nThis action CANNOT be undone!\n\nType \"DELETE\" below to confirm.")
                .setPositiveButton("Delete My Account", (d, w) -> {
                    new AlertDialog.Builder(this)
                        .setTitle("Final Confirmation")
                        .setMessage("Are you ABSOLUTELY sure?\n\nYour account data, friends, stats, and everything will be permanently deleted.")
                        .setPositiveButton("Yes, Delete Everything", (d2, w2) -> deleteAccount())
                        .setNegativeButton("Cancel", null)
                        .show();
                })
                .setNegativeButton("Cancel", null)
                .show();
        });

        // Notification preferences (stored locally via SharedPreferences)
        switchNotifications.setOnCheckedChangeListener((buttonView, isChecked) -> {
            // Save preference
            getSharedPreferences("notif_prefs", MODE_PRIVATE)
                .edit()
                .putBoolean("notifications_enabled", isChecked)
                .apply();
        });

        switchChatNotifs.setOnCheckedChangeListener((buttonView, isChecked) -> {
            getSharedPreferences("notif_prefs", MODE_PRIVATE)
                .edit()
                .putBoolean("chat_notifs_enabled", isChecked)
                .apply();
        });

        switchScheduleNotifs.setOnCheckedChangeListener((buttonView, isChecked) -> {
            getSharedPreferences("notif_prefs", MODE_PRIVATE)
                .edit()
                .putBoolean("schedule_notifs_enabled", isChecked)
                .apply();
        });
    }

    private void deleteAccount() {
        String uid = FirebaseAuth.getInstance().getUid();
        if (uid == null) {
            Toast.makeText(this, "Not logged in", Toast.LENGTH_SHORT).show();
            return;
        }

        FirebaseFirestore db = FirebaseFirestore.getInstance();

        // Remove from players collection
        db.collection("players").document(uid).delete()
            .addOnSuccessListener(v -> {
                // Remove friend requests
                db.collection("friend_requests").whereEqualTo("fromUserId", uid).get()
                    .addOnSuccessListener(reqs -> {
                        for (com.google.firebase.firestore.DocumentSnapshot req : reqs.getDocuments()) {
                            req.getReference().delete();
                        }
                        finishDeletion();
                    });
            })
            .addOnFailureListener(e -> {
                Toast.makeText(this, "Failed to delete account", Toast.LENGTH_SHORT).show();
            });
    }

    private void finishDeletion() {
        // Sign out and clear local data
        FirebaseAuth.getInstance().signOut();
        getSharedPreferences("notif_prefs", MODE_PRIVATE).edit().clear().apply();

        Toast.makeText(this, "Account deleted successfully", Toast.LENGTH_LONG).show();

        new AlertDialog.Builder(this)
            .setTitle("Account Deleted")
            .setMessage("Your account has been permanently deleted.\n\nReturning to login...")
            .setPositiveButton("OK", (d, w) -> {
                Intent intent = new Intent(this, LoginActivity.class);
                intent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
                startActivity(intent);
                finish();
            })
            .setCancelable(false)
            .show();
    }
}

File: app/src/main/java/com/zerostress/manager/SettingsActivity.kt
package com.zerostress.manager

import android.content.Intent
import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Modifier
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.firestore.FirebaseFirestore
import com.zerostress.manager.ui.theme.*

class SettingsActivity : ComponentActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContent { ZeroStressTheme { SettingsScreen() } }
    }
}

@Composable
fun SettingsScreen() {
    val context = androidx.compose.ui.platform.LocalContext.current
    var notifications by remember { mutableStateOf(true) }
    var chatNotifs by remember { mutableStateOf(true) }
    var scheduleNotifs by remember { mutableStateOf(true) }
    var showDeleteDialog by remember { mutableStateOf(false) }

    if (showDeleteDialog) {
        AlertDialog(
            onDismissRequest = { showDeleteDialog = false },
            title = { Text("⚠️ Delete Account") },
            text = { Text("This will permanently delete your account and all data. This action CANNOT be undone!") },
            confirmButton = {
                TextButton(onClick = {
                    val uid = FirebaseAuth.getInstance().uid
                    if (uid != null) {
                        FirebaseFirestore.getInstance().collection("players").document(uid).delete()
                        FirebaseAuth.getInstance().currentUser?.delete()
                        FirebaseAuth.getInstance().signOut()
                        context.startActivity(Intent(context, LoginActivity::class.java))
                        (context as? ComponentActivity)?.finish()
                    }
                    showDeleteDialog = false
                }) { Text("Delete Everything", color = Color.Red) }
            },
            dismissButton = { TextButton(onClick = { showDeleteDialog = false }) { Text("Cancel") } }
        )
    }

    Column(modifier = Modifier.fillMaxSize().padding(16.dp)) {
        Text("⚙ Settings", fontSize = 22.sp, fontWeight = FontWeight.Bold, color = Color.White)
        Spacer(modifier = Modifier.height(16.dp))

        Card(shape = RoundedCornerShape(12.dp), colors = CardDefaults.cardColors(containerColor = ZSCard)) {
            Column(modifier = Modifier.padding(16.dp)) {
                Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
                    Text("Notifications", color = Color.White); Switch(checked = notifications, onCheckedChange = { notifications = it })
                }
                Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
                    Text("Chat Notifications", color = Color.White); Switch(checked = chatNotifs, onCheckedChange = { chatNotifs = it })
                }
                Row(modifier = Modifier.fillMaxWidth(), horizontalArrangement = Arrangement.SpaceBetween) {
                    Text("Schedule Notifications", color = Color.White); Switch(checked = scheduleNotifs, onCheckedChange = { scheduleNotifs = it })
                }
            }
        }

        Spacer(modifier = Modifier.height(12.dp))

        Card(shape = RoundedCornerShape(12.dp), colors = CardDefaults.cardColors(containerColor = ZSCard)) {
            Column(modifier = Modifier.padding(16.dp)) {
                Button(onClick = { /* clear cache */ }, modifier = Modifier.fillMaxWidth()) { Text("Clear Cache") }
                Spacer(modifier = Modifier.height(8.dp))
                OutlinedButton(onClick = {
                    androidx.appcompat.app.AlertDialog.Builder(context)
                        .setTitle("ZERO STRESS")
                        .setMessage("Version 4.0\n\nPerformance & Leaderboard Manager\n\nBuilt with Firebase + Kotlin + Jetpack Compose")
                        .setPositiveButton("OK", null)
                        .show()
                }, modifier = Modifier.fillMaxWidth()) { Text("About") }
            }
        }

        Spacer(modifier = Modifier.height(12.dp))

        Button(
            onClick = {
                FirebaseAuth.getInstance().signOut()
                context.startActivity(Intent(context, LoginActivity::class.java))
                (context as? ComponentActivity)?.finish()
            },
            modifier = Modifier.fillMaxWidth(),
            colors = ButtonDefaults.buttonColors(containerColor = ZSWarning)
        ) { Text("Logout", color = Color.Black) }

        Spacer(modifier = Modifier.height(8.dp))

        Button(
            onClick = { showDeleteDialog = true },
            modifier = Modifier.fillMaxWidth(),
            colors = ButtonDefaults.buttonColors(containerColor = ZSDanger)
        ) { Text("Delete Account") }
    }
}
